C/cufinufft/build_tarballs.jl

```diff
@@ -0,0 +1,83 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+using BinaryBuilderBase
+include(joinpath(@__DIR__, "..", "..", "fancy_toys.jl"))
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
+# Build script for the CUDA part of FINUFFT
+
+# Builds for all compatible CUDA platforms, but without microarchitecture expansion (not
+# needed for CUDA cuda, and would produce a giant amount of artifacts)
+name = "cufinufft"
+version = v"2.2.0"
+
+commit_hash = "51892059a4b457a99a2569ac11e9e91cd2e289e7";
+preferred_gcc_version=v"11"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/flatironinstitute/finufft.git", commit_hash)
+]
+
+# Build script: cufinufft, all possible archs available for each CUDA version
+# - CMake toolchain looks for compiler in CUDA_PATH/bin/nvcc
+#   and libs in CUDA_PATH/lib64, so create link.
+script = raw"""
+cd $WORKSPACE/srcdir/finufft*/
+
+export CUDA_PATH="$prefix/cuda"
+ln -s $prefix/cuda/lib $prefix/cuda/lib64
+
+mkdir build && cd build
+cmake .. \
+    -DCMAKE_PREFIX_PATH="${prefix}" \
+    -DCMAKE_INSTALL_PREFIX="${prefix}" \
+    -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+    -DFINUFFT_FFTW_SUFFIX="" \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DFINUFFT_USE_CPU=OFF \
+    -DFINUFFT_USE_CUDA=ON \
+    -DCMAKE_CUDA_ARCHITECTURES="${CUDA_ARCHS}"
+cmake --build . --parallel $nproc
+cmake --install .
+"""
+
+# Build for all supported CUDA > v11
+platforms = CUDA.supported_platforms(min_version=v"11.0")
+# Cmake toolchain breaks on aarch64, so only x86_64 for now
+filter!(p -> p["arch"]=="x86_64", platforms)
```

Comment:
Just slightly nicer:
```suggestion
filter!(p -> arch(p)=="x86_64", platforms)
```